### PR TITLE
SPUI-1004/1017 > Add automation country code in phone number

### PR DESCRIPTION
**Pull request by DenisMartych**

### CODE REVIEW

File: lib/app/shared/helpers/country_code_by_user_register.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'package:simple_kit/simple_kit.dart';
+
+SPhoneNumber? countryCodeByUserRegister(String countryOfRegistration) {
+  SPhoneNumber? phoneNumber;
+
+  for (var i = 0; i < sPhoneNumbers.length; i++) {
+    if (sPhoneNumbers[i].alphaCode.toLowerCase() ==
```
Comment (eli1stark) [inline lib/app/shared/helpers/country_code_by_user_register.dart:7]: investigate alphaCode

File: lib/app/shared/helpers/country_code_by_user_register.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import 'package:simple_kit/simple_kit.dart';
+
+SPhoneNumber? countryCodeByUserRegister(String countryOfRegistration) {
+  SPhoneNumber? phoneNumber;
+
+  for (var i = 0; i < sPhoneNumbers.length; i++) {
```
Comment (eli1stark) [inline lib/app/shared/helpers/country_code_by_user_register.dart:6]: for (final number in sPhoneNumbers)

File: lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart [hunk 3]
```diff
@@ -101,4 +105,23 @@ class SetPhoneNumberNotifier extends StateNotifier<SetPhoneNumberState> {
       activeDialCode: code,
     );
   }
+
+  void _registerCountryUser() {
+    final userInfo = read(userInfoNotipod);
+
+    if (userInfo.countryOfRegistration.isNotEmpty) {
```
Comment (eli1stark) [inline lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart:112]: move to countryCodeByUserRegister

File: lib/app/shared/helpers/country_code_by_user_register.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import 'package:simple_kit/simple_kit.dart';
+
+SPhoneNumber? countryCodeByUserRegister(String countryOfRegistration) {
```
Comment (eli1stark) [inline lib/app/shared/helpers/country_code_by_user_register.dart:3]: add Reader `read`

File: lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart [hunk 5]
```diff
@@ -101,4 +105,23 @@ class SetPhoneNumberNotifier extends StateNotifier<SetPhoneNumberState> {
...
+      final phoneNumber = countryCodeByUserRegister(
+        userInfo.countryOfRegistration,
```
Comment (eli1stark) [inline lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart:114]: replace by Reader `read`

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/shared/helpers/country_code_by_user_register.dart (hunk 1 above)
Comment (DenisMartych) [reply lib/app/shared/helpers/country_code_by_user_register.dart:7]: fix, replace on isoCode

File: lib/app/shared/helpers/country_code_by_user_register.dart (hunk 2 above)
Comment (DenisMartych) [reply lib/app/shared/helpers/country_code_by_user_register.dart:6]: done

File: lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart (hunk 3 above)
Comment (DenisMartych) [reply lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart:112]: done

File: lib/app/shared/helpers/country_code_by_user_register.dart (hunk 4 above)
Comment (DenisMartych) [reply lib/app/shared/helpers/country_code_by_user_register.dart:3]: done

File: lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart (hunk 5 above)
Comment (DenisMartych) [reply lib/app/shared/features/set_phone_number/notifier/set_phone_number_notifier.dart:114]: done

**Pull request merged**